feat: CMS GC enabled by default (#5558)
JetBrains/kotlin-web-site: docs/topics/native/native-memory-manager.md

## Garbage collector

Kotlin/Native's garbage collector (GC) algorithm is constantly evolving. Currently, it functions as a concurrent mark
and sweep (CMS) collector that does not separate the heap into generations.

The GC is executed on a separate thread and started based on the memory pressure heuristics or by a timer. Alternatively,
it can be [called manually](#enable-garbage-collection-manually).

The GC processes the mark queue on several threads in parallel, including application threads, the GC thread,
and optional marker threads. Application threads and at least one GC thread participate in the marking process.

> By default, the marking phase runs concurrently with application threads, which decreases the GC pause time.

Reviewer: ```suggestion
By default, the marking phase runs concurrently with application threads, which reduces the GC pause time.
```

Reviewer: (suggestion) imho both works so whichever you prefer

> You can monitor GC performance [through logs](#monitor-gc-performance).

Reviewer: ```suggestion
You can monitor GC performance through the [GC logs](#monitor-gc-performance).
```

Reviewer: again both work, but I see we use this for example in[ Integration with Swift](https://kotlinlang.org/docs/native-arc-integration.html#objective-c-objects-lifecycle) and I think it's a bit more specific

When the marking phase is completed, the GC processes weak references and nullifies reference points to an unmarked object.
By default, weak references are processed concurrently to decrease the GC pause time.

> If you face problems with CMS, switch back to parallel mark concurrent sweep (PMCS) setup.

Reviewer: ```suggestion
If you encounter problems with CMS, switch back to the parallel mark concurrent sweep (PMCS) setup.
```
